src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -141,6 +141,12 @@ public class CheckForPoison : Task
 
         private const string PoisonMarker = "POISONED";
 
+        private class CandidateFileEntry
+        {
+            public string ExtractedPath { get; set; }
+            public string DisplayPath { get; set; }
+        }
```

Comment:
Consider using a record type:

```suggestion
        private record CandidateFileEntry(string ExtractedPath, string DisplayPath);
```

Reply:
Updated in this commit: https://github.com/dotnet/installer/pull/17438/commits/11232daadf922b0592118dd9f588e6e77854f66b